### ISSUE TITLE
Difference between mutable and immutable collection

### ISSUE THREAD

**Issue by anaelChardan** [Novactive/NovaCollection#16]
IMHO, it's quite difficult to remember which function is mutable and which function is not

It could be nice that Collection can be split into MutableCollection and ImmutableCollection with the same name in functions (due to an interface contract).

**Comment by Plopix** [comment]
we discussed that. we have decided not do it. Forcing the selection of MutableCollection or ImmutableCollection might not be the "PHP way". We could play with traits and have Collection and ImmutableCollection. We could reconsider, it is a good subject.

**Comment by anaelChardan** [comment]
Yes, I know, but I'm always frustrated as I do not the name of functionnal methods when it is mutable. When we come from Java world or Scala world, we're aware about that :) http://docs.scala-lang.org/overviews/collections/overview.html
